src/mini_batch.jl

```diff
@@ -26,99 +33,100 @@ function kmeans!(alg::MiniBatch, X, k;
     N = zeros(T, k)
 
     # Initialize nearest centers for both batch and whole dataset labels
-    final_labels = Vector{Int}(undef, ncol)  # dataset labels
-
     converged = false
     niters = 0
     counter = 0
     J_previous = zero(T)
     J = zero(T)
+    totalcost = zero(T)
 
```

Comment:
You can extract `labels` from `containers` with
```
labels = containers.labels
```

and use this aliased form in all other calculations. Containers are useful, when you split this huge monolithic function into smaller parts, it's easier to pass single parameter, instead of a few. But inside the function containers content should be extracted, it's just easier to work with. You can see `Lloyd` and other algorithms implementation as an example.

Reply:
Unfortunately, aliasing seem to be allocating. Will need to look into it further.

Reply:
No it doesn't :-) It can not. Problem is somewhere else.